base/linalg/bidiag.jl

```diff
@@ -224,9 +224,43 @@ end
 /(A::Bidiagonal, B::Number) = Bidiagonal(A.dv/B, A.ev/B, A.isupper)
 ==(A::Bidiagonal, B::Bidiagonal) = (A.dv==B.dv) && (A.ev==B.ev) && (A.isupper==B.isupper)
 
-SpecialMatrix = Union{Bidiagonal, SymTridiagonal, Tridiagonal, AbstractTriangular}
+SpecialMatrix = Union{SymTridiagonal, AbstractTriangular}
 *(A::SpecialMatrix, B::SpecialMatrix)=full(A)*full(B)
 
+function A_mul_B!(C::AbstractMatrix, A::Bidiagonal, B::Bidiagonal)
+    if size(B,1) < 3
+        return A_mul_B!(C, A, full(B))
+    else
+        return A_mul_B!(C, Tridiagonal(A), Tridiagonal(B))
+    end
+end
+
+for MAT in [Tridiagonal, AbstractTriangular, AbstractMatrix]
+    q = quote function A_mul_B!(C::AbstractMatrix, A::$(MAT), B::Bidiagonal)
+            if size(B,1) < 3
+                return A_mul_B!(C, A, full(B))
+            else
+                return A_mul_B!(C, A, Tridiagonal(B))
+            end
+        end
+    end
+    eval(q)
```

Comment:
Are you sure? I was under the impression I need to do it like this to resolve definition ambiguities.